multidim-interop/rust/src/bin/testplan_0500.rs

```diff
@@ -1,144 +1,148 @@
+use std::collections::HashSet;
+use std::env;
+use std::time::Duration;
+
 use anyhow::{Context, Result};
 use async_trait::async_trait;
 use futures::{AsyncRead, AsyncWrite, StreamExt};
+use libp2p::core::muxing::StreamMuxerBox;
 use libp2p::core::transport::Boxed;
 use libp2p::core::upgrade::EitherUpgrade;
-use libp2p::noise::NoiseOutput;
-use libp2p::tls::TlsStream;
-use libp2p::{core::muxing::StreamMuxerBox, swarm::derive_prelude::EitherOutput};
+use libp2p::swarm::{keep_alive, NetworkBehaviour, SwarmEvent};
+use libp2p::websocket::WsConfig;
+use libp2p::{
+    core, identity, mplex, noise, ping, webrtc, yamux, Multiaddr, PeerId, Swarm, Transport as _,
+};
 use libp2pv0500 as libp2p;
-use libp2pv0500::swarm::SwarmEvent;
-use libp2pv0500::websocket::WsConfig;
-use libp2pv0500::*;
-use std::collections::HashSet;
-use std::env;
-use std::time::Duration;
-use testplan::{run_ping_redis, PingSwarm};
+use testplan::{run_ping, Muxer, PingSwarm, SecProtocol, Transport};
 
 fn build_builder<T, C>(
```

Comment:
This function doesn't have much value because the type constraints are so insane. You are likely better off by just inlining things and accepting the duplication between the different invocations.

Reply:
I still feel it's more lines of code and repetition, but feel free to submit a PR addressing that Thomas